app/models/data.py

```diff
@@ -252,6 +252,27 @@ def totalTestResults(self):
     def parse_str_to_date(date_input):
         return parser.parse(str(date_input), ignoretz=True).date()
 
+    @staticmethod
+    def valid_fields_checker(candidates):
+        '''
+        dict[string] -> ([string], [string])
+        Gets a list of field names and returns a tuple of (valid_fields, unknown_fields).
+
+        If valid_fields is empty, then this list contains no data for this object.
+        If unknown_fields is not-empty, then this list contains extra fields
+        that mean nothing for this object, but we might want to alert on this.
+        In the valid fields, we exclude state, date and batchId
+        '''
+        mapper = class_mapper(CoreData)
+        keys = mapper.attrs.keys()
+
+        candidate_set = set(candidates)
+        key_set = set(keys)
+        unknowns = candidate_set - key_set
+        valid = candidate_set & key_set
+        valid = valid - {'state', 'date', 'batchId'}
```

Comment:
to avoid using string constants you can also get these as the primary key names - `{x.name for x in mapper.primary_key}`